RFCs/0028-BlameCollector-Hang-Detection.md

```diff
@@ -1,44 +1,88 @@
 
-```
+Add the required settings as shown below in the .runsettings file sample below:
+
+```xml
 <RunSettings>
-    <DataCollectionRunSettings>
-        <DataCollectors>
-            <DataCollector friendlyName="blame" enabled="True">
-                <Configuration>
-                    <CollectDumpOnTestSessionHang TestTimeout="300000" DumpType="mini" />
-                    <ResultsDirectory>%AGENT_TEMPDIRECTORY%\TestResults</ResultsDirectory>
-                </Configuration>
-            </DataCollector>
-        </DataCollectors>
-    </DataCollectionRunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="blame" enabled="True">
+        <Configuration>
+          <CollectDumpOnTestSessionHang TestTimeout="300000" DumpType="mini" />
+          <ResultsDirectory>%AGENT_TEMPDIRECTORY%</ResultsDirectory>
+        </Configuration>
+      </DataCollector>
+    </DataCollectors>
+  </DataCollectionRunSettings>
 </RunSettings>
 ```
 
-Note: This works along with (but can also be used independent of) the existing collect dump option introduced in https://github.com/microsoft/vstest-docs/blob/master/RFCs/0024-Blame-Collector-Options.md
+Note: This works along with (but can also be used independent of) the existing collect dump option introduced in [RFC 0024](0024-Blame-Collector-Options.md).
 
 ## Supported options
-DumpType: If you choose to collect a full process dump. It takes values mini/full. By default, a mini dump will be created.
-TestTimeout: Duration of inactivity in milliseconds (no test events from the test host) after which the data collector assumes a hang has occurred and proceeds to collect a dump and kill the test host process.
-ResultsDirectory: Location to temporarily store the collected process dump
+
+|Option|Description|
+|--|--|
+`DumpType`|If you choose to collect a full process dump. It takes values `mini`/`full`. By default, a mini dump will be created.
+`TestTimeout`|Duration of inactivity in milliseconds (no test events from the test host) after which the data collector assumes a hang has occurred and proceeds to collect a dump and kill the test host process.
+`ResultsDirectory`|The path must exist, but it is not used. Test results go to a `TestResults` folder under the test project directory when `dotnet test` is used locall, or under `$(Agent.TempDirectory)\<guid>` when the `DotNetCoreCLI` task is used on Azure Pipelines.
+
+## Invoking the tests
+
+Add the `--settings path-to\test.runsettings` parameter to `dotnet test` or `vstest.console.exe`.
+Note that this will cause non-Windows agents to fail, so only add this switch on Windows agents.
+
+## Outputs
+
+The `Sequence_guid.xml` and .dmp files will be placed under the test project source directory in a `TestResults\<guid>\` folder.
+
+The `DotNetCoreCLI` task will try to upload these files as attachments, but any dump file exceeding 75MB will fail (and full dumps are almost surely going to exceed this limit).
```

Comment:
This limitation is only with the dotnet task, won't be a problem with the vstest task.

I'd prefer if we actually make 2 separate sections one for vstest and one for dotnet test to make sure the limitations are called out accordingly

Reply:
Sounds good.